olcPixelGameEngine.h

```diff
@@ -2332,6 +2336,99 @@ namespace olc
 		}
 	}
 
+
+  void PixelGameEngine::DrawPolygon(const std::vector<olc::vi2d>& points, olc::Pixel p)
+  {
+    // Maybe nothing to do?
+    if (points.size() <= 0) 
+      return;
+
+    // Draw every line and close the polygon
+    for (int i=0; i < points.size(); i++) {
+      DrawLine(points[i], points[(i+1) % points.size()], p);
+    }
+  }  
+
+  void PixelGameEngine::FillPolygon(const std::vector<olc::vi2d>& points, olc::Pixel p)
+  {
+    // Maybe nothing to do?
+    if (points.size() <= 0) 
+      return;
+
+    // Find the min and max Y-values
+    float starty = (float)vScreenSize.y;
+    float stopy = 0.f;
+    for (auto &p: points)
+    {
+      if (p.y < starty) starty = p.y;
+      if (p.y > stopy)  stopy  = p.y;
+    }
+
+    // Run through every scanline containing the polygon and convert to lines
+    std::vector<int32_t> nodex;
+    for (int32_t y = (int32_t)starty; y < (int32_t)stopy; y++)
+    {
+      //  Build a list of nodes.
+      nodex.clear();
+      int j = points.size() - 1;
+      for (int i=0; i<points.size(); i++)
+      {
+        float pxi = points[i].x; 
+        float pyi = points[i].y; 
+        float pxj = points[j].x; 
+        float pyj = points[j].y; 
+
+        if (  (pyi < (float)y && pyj >= (float)y) || (pyj < (float)y && pyi >= (float)y) )
+        {
+          nodex.push_back((int32_t)(pxi + (y - pyi) / (pyj - pyi) * (pxj - pxi)));
+        }
+        j=i;
+      }
+
+      // Maybe no scanline goes through polygon (should not occur, but just to be sure...)
+      if (nodex.size() <= 0)
+        continue;
+
+      //  Sort the nodes, via a simple “Bubble” sort.
+      int i=0;
+      float swap;
+      while (i < nodex.size() - 1) 
+      {
+        if (nodex[i] > nodex[i+1]) 
+        {
+          swap = nodex[i]; 
+          nodex[i] = nodex[i+1]; 
+          nodex[i+1] = swap; 
+          if (i > 0) 
+            i--; 
+        }
+        else 
+        {
+          i++; 
+        }
+      }

```

Comment:
I see algorithm is included so might as well use `std::sort`

Reply:
yeah, I didn't want to rely on other stuff